library/src/updater.rs

```diff
@@ -421,41 +421,51 @@ pub fn report_launch_start() -> anyhow::Result<()> {
         // Make that patch the "booted" patch.
         state.activate_current_patch()?;
         state.save()
-    })
+    })?
 }
 
 /// Report that the current active path failed to launch.
 /// This will mark the patch as bad and activate the next best patch.
-pub fn report_launch_failure() -> Result<(), UpdateError> {
+pub fn report_launch_failure() -> anyhow::Result<()> {
```

Comment:
UpdateError was a dream.  Probably can die now that we've teched up to anyhow.

Reply:
Filed https://github.com/shorebirdtech/updater/issues/19 – I think we'll want to overhaul our error handling more generally as part of this.